Music app/app/src/main/java/com/example/musicapp/NotificationPlaying/MusicService.java

```diff
@@ -0,0 +1,209 @@
+            mediaPlayer.release();
+            if (musicFiles != null) {
+                createMediaPlayer(position);
+                mediaPlayer.start();
+            }
+        } else {
+            createMediaPlayer(position);
+            mediaPlayer.start();
+        }
+    }
+
+    public void start() {
+        mediaPlayer.start();
+    }
+
+    public boolean isPlaying() {
+        return mediaPlayer.isPlaying();
+    }
+
+    public void stop() {
+        mediaPlayer.stop();
+    }
+
+    public void release() {
+        mediaPlayer.release();
+    }
+
+    public int getDuration() {
+        return mediaPlayer.getDuration();
+    }
+
+    public void seekTo(int position) {
+        mediaPlayer.seekTo(position);
+    }
+
+    public int getCurrentPosition() {
+        return mediaPlayer.getCurrentPosition();
+    }
+
+    public void createMediaPlayer(int positionInner) {
+        position = positionInner;
+        uri = musicFiles.get(position).getUri();
+        mediaPlayer = MediaPlayer.create(getBaseContext(), uri);
+    }
+
+    public void OnCompleted() {
+        mediaPlayer.setOnCompletionListener(this);
+    }
+
+    @Override
+    public void onCompletion(MediaPlayer mp) {
+        if (actionPlaying != null) {
+            actionPlaying.nextBtnClicked();
+            if (mediaPlayer != null){
+                createMediaPlayer(position);
+                mediaPlayer.start();
+                OnCompleted();
+            }
+        }
+    }
+
+    public void setCallBack(ActionPlaying actionPlaying) {
+        this.actionPlaying = actionPlaying;
+    }
+
+    public void showNotification(int playPauseBtn) {
+        Intent prevIntent = new Intent(this, NotificationReceiver.class)
+                .setAction(ApplicationClass.ACTION_PREVIOUS);
+        PendingIntent prevPending = PendingIntent
+                .getBroadcast(this, 0, prevIntent,
+                        PendingIntent.FLAG_UPDATE_CURRENT);
+
+        Intent pauseIntent = new Intent(this, NotificationReceiver.class)
+                .setAction(ApplicationClass.ACTION_PLAY);
+        PendingIntent pausePending = PendingIntent
+                .getBroadcast(this, 0, pauseIntent,
+                        PendingIntent.FLAG_UPDATE_CURRENT);
+
+        Intent nextIntent = new Intent(this, NotificationReceiver.class)
+                .setAction(ApplicationClass.ACTION_NEXT);
+        PendingIntent nextPending = PendingIntent
+                .getBroadcast(this, 0, nextIntent,
+                        PendingIntent.FLAG_UPDATE_CURRENT);
+
+        byte[] picture = MusicAdapter.getAlbumArt(musicFiles.get(position).getPath());
+        Bitmap thumb;
+        if (picture != null) {
+            thumb = BitmapFactory.decodeByteArray(picture, 0, picture.length);
+        } else {
+            thumb = BitmapFactory.decodeResource(getResources(), R.drawable.musicimage);
+        }
+        Notification notification = new NotificationCompat.Builder(this,
```

Comment:
it's kind of like a stream, isn't it?

Reply:
Yes, this is some kind of builder class. It is extremely strong because it allows such an easy setup.

Reply:
I can't remember why there had to be two, but NotificationManager just requires it.